### ISSUE TITLE
Issues of get_windows Function

### ISSUE THREAD

**Issue by erinqhu** [Lif3line/nina_helper_package_mk2#4]
![image](https://user-images.githubusercontent.com/27153832/153097447-c8c7dbe4-24b8-4e55-889c-56d62479fb90.png)
Alireaza ([asarmadi](https://github.com/parasgulati8/human_identification/commits?author=asarmadi)) and I found that the above part of the get_windows function was not 100% correct. The above snippet should be replaced by the code snippet below to get rid of the rows (windows) where sEMG values are all zeros.
![image](https://user-images.githubusercontent.com/27153832/153097762-ea07b87e-bb5d-46df-8d56-528afb931dc0.png)
Please let us know whether this observation is correct.
Thanks.

**Comment by Lif3line** [comment]
Hi @erinqhu, you're right the function does allow 0 regions to be valid, however, this is deliberate. The aim is to preserve the continuity of the time stream, so that the resultant arrays can be directly compared to the original data. Therefore it is implicitly assumed that any value in `movements` is valid, including 0. 

If you'd like to submit a PR to make the above behavior optional then please do, but I think the default behavior should still be to retain the dimensionality of the data stream.

**Comment by erinqhu** [comment]
> Hi @erinqhu, you're right the function does allow 0 regions to be valid, however, this is deliberate. The aim is to preserve the continuity of the time stream, so that the resultant arrays can be directly compared to the original data. Therefore it is implicitly assumed that any value in `movements` is valid, including 0.
> 
> If you'd like to submit a PR to make the above behavior optional then please do, but I think the default behavior should still be to retain the dimensionality of the data stream.

Hi @Lif3line, thanks for your reply. These zero windows usually exist at the beginning of each repetition since window_ends are used in the slicing. I hope these zero windows will not affect much of the performance of any model. I'll go ahead and submit a PR to make the behavior as an option in the get_windows function.

**Comment by Lif3line** [comment]
Yeah, the process was part of an effort to learn well-defined gesture boundaries being determined by a classifier from a continual stream, hence preservation of the time-stream. They'll definitely affect model performance though naturally the magnitude will depend on the data, model, and metric specifics. I've spent some time exploring how the edge effects of movement transitions affects performance and I'd definitely recommend trying with and without, especially the mixed cases where your training set uses one method and your validation and/or testing uses the other. I've found that tends to highlight how important it is for the given application.
